src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,2857 @@
+            //     ReadOnlySpan<object> M6(string?[] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[]", "System.ReadOnlySpan<object>").WithLocation(10, 47),
+            // (13,47): warning CS8619: Nullability of reference types in value of type 'object?[]' doesn't match target type 'string[]'.
+            //     ReadOnlySpan<string> M8(object?[] arg) => (ReadOnlySpan<string>)arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(ReadOnlySpan<string>)arg").WithArguments("object?[]", "string[]").WithLocation(13, 47)
+        };
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        CreateCompilationWithSpan(source).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis_NestedArrays()
+    {
+        var source = """
+            #nullable enable
+            using System;
+            class C
+            {
+                Span<string[]> M1(string[][] arg) => arg;
+                Span<string[]> M2(string?[][] arg) => arg;
+                Span<string?[]> M3(string[][] arg) => arg;
+                Span<string?[]> M4(string?[][] arg) => arg;
+
+                Span<string[]> M5(string?[][] arg) => (Span<string[]>)arg;
+            }
+            """;
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (6,43): warning CS8619: Nullability of reference types in value of type 'string?[][]' doesn't match target type 'string[][]'.
+            //     Span<string[]> M2(string?[][] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[][]", "string[][]").WithLocation(6, 43),
+            // (10,43): warning CS8619: Nullability of reference types in value of type 'string?[][]' doesn't match target type 'string[][]'.
+            //     Span<string[]> M5(string?[][] arg) => (Span<string[]>)arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(Span<string[]>)arg").WithArguments("string?[][]", "string[][]").WithLocation(10, 43));
+
+        var expectedDiagnostics = new[]
+        {
+            // (6,43): warning CS8619: Nullability of reference types in value of type 'string?[][]' doesn't match target type 'Span<string[]>'.
+            //     Span<string[]> M2(string?[][] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[][]", "System.Span<string[]>").WithLocation(6, 43)
+        };
+
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        CreateCompilationWithSpan(source).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Fact]
+    public void Conversion_Array_ReadOnlySpan_Implicit_NullableAnalysis_NestedArrays()
```

Comment:
Confused again. I'd have expected warnings on `M3` and `M7` in both scenarios.

Reply:
> in both scenarios

C# 12 scenario isn't changed, so that might be a pre-existing bug.

Reply:
I've investigated more and there are pre-existing inconsistencies which I think are out of scope for this feature:
- HasExplicitArrayConversion does *not* consider nullability of element types
- HasImplicitConversionFromArray does consider nullability of element types